source/rust_verify/tests/recursion.rs

```diff
@@ -1215,3 +1215,29 @@ test_verify_one_file! {
         }
     } => Err(e) => assert_one_fails(e)
 }
+
+test_verify_one_file! {
+    #[test] decreases_on_non_recursive_generic_datatype verus_code! {
```

Comment:
can you please add `_regression_315` to the test name so we can track that this is a regression test?
Thanks!